util/create_fencing_agent.sh

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+#
+#######################################################################################################################################################
+#
+# This script simplifies the user interaction with Azure to create a fencing
+# agent service principal and grant the required permissions to manage resources.
+# This script is essentially wrapping the process described here:
+# https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/sap/high-availability-guide-suse-pacemaker#create-azure-fence-agent-stonith-device
+#
+#######################################################################################################################################################
+
+# exit immediately if an unset variable is used
+set -o nounset
+
+# import common functions that are reused across scripts
+source util/common_utils.sh
+
+# import functions to create Service Principle
+
+# source util/create_service_principal.sh
+
+# name of the script where the auth info will be saved
+readonly auth_script="set-clustering-auth-${1-}.sh"
+
+# link for service principal help
+readonly sp_link='https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli'
+
+function main()
+{
+	check_command_line_arguments "$@"
+
+	local SID="$1"
+
+	create_service_principal_script "${SID}"
+}
+
+
+
+function check_command_line_arguments()
+{
+	local args_count=$#
+
+	# Check there's just a single argument provided
+	if [[ ${args_count} -ne 1 ]]; then
+		error_and_exit "You must specify a single command line argument for the SAP SID"
+	fi
+}
+
+
+function create_service_principal_script()
+{
+	local service_principal_name="fencing-agent-$1"
+
+	check_auth_script_does_not_exist
+
+	echo "Creating Azure Service Principal: ${service_principal_name}..."
+
+	# ensure newlines in variable values are preserved
+	# backup current value to restore to afterwards
+	local ifs_backup="${IFS}"
+	IFS=$'\n'
+
+	# create the service principal and capture the output
+	sp_details=$(az ad sp create-for-rbac --name "${service_principal_name}")
+	local sp_details=$(az ad sp create-for-rbac --name "${service_principal_name}")
+	local sp_creation_status=$?=$?
```

Comment:
This is the reason your script is erroring when run.

```suggestion
ocal sp_creation_status=$?
```